### ISSUE TITLE
Self Contained HTML File Export Can't Display Images With URL Encoded Paths

### ISSUE THREAD

**Issue by jbratu** [RickStrahl/MarkdownMonster#560]
In Markdown Monster 1.18.10 with the Dharkan theme, my image names that contain spaces don't work properly. When I drag the image with spaces in the name into Markdown Monster it URL encodes the spaces to `%20`. These images do not display properly when the `File -> Save To -> Save to HTML` menu is selected and the *Save as Type* is set to `Self contained Html File with embedded Styles and Images` is selected.  The image is not encoded into the produced HTML file and does not properly display.

To reproduce the issue:
1. Make an *assets* folder relative to the markdown document
2. Place an image in the folder. i.e. `my Network layout.png`
3. Copy the image and remove the spaces so the name is `myNetworklayout.png`
4. Drag each image into the Markdown document one at a time. It should look something like:
```
Encoded

![](assets/my%20Network%20layout.png)

No Encoding Required

![](assets/myNetworklayout.png)
```

Both images should appear in the preview.
5. Select `File -> Save To -> Save to HTML` on the menu and the *Save as Type* set to `Self contained Html File with embedded Styles and Images`.
6. Save the HTML file into a folder other than where the original markdown document was located.

The saved HTML file will show the image under *No Encoding Required* but the *Encoded* message will not display.

**Comment by RickStrahl** [comment]
I can't duplicate what you describe.

Here's what I get with encoded content. Original content in MM on left, and exportedfile on right:

![image](https://user-images.githubusercontent.com/1374013/63375906-7ed9bc00-c341-11e9-927d-db6f6450ac67.png)

File is local on disk with images in a relative folder - looks the same as what you're metnioning above.

**Comment by RickStrahl** [comment]
Ok so on closer examination I see the problem - it works above because the relative link is still valid, but the file was not properly imported.

The bug is that for local resources the URL is not properly UrlDecoded so that the file to read and embed is not found.

**Comment by RickStrahl** [comment]
Fixed in e2af8a511f375b18384233f485e41db45e546d32
